tools/clang/test/CodeGenDXIL/hlsl/linalg/mat-vec-mul-transpose-errors.hlsl

```diff
@@ -0,0 +1,29 @@
+// RUN: %dxc -T lib_6_9 -enable-16bit-types %s | FileCheck %s
+
+ByteAddressBuffer Buf;
+
+export float4 Test1(vector<float, 4> Input) {
+  using namespace dx::linalg;
+
+  MatrixRef<DATA_TYPE_FLOAT16, 4, 4, MATRIX_LAYOUT_ROW_MAJOR, true> Matrix = {
+      Buf, 0, 0};
+
+  // clang-format off
+  // CHECK: error: something about transposing not supported for rowmajor / colmajor layouts
```

Comment:
presumably this check fails?

Reply:
Yes, it's a placeholder for now. There should be other placeholders too. Seeking feedback on whether this should be an error at the sema-level, or whether this should be left as a validation error. If Sema, I can come up with a better message, and change the code to emit the modified error message. If left to the validator, I'd break this test out to a new file and test for the validator error.

Reply:
I don't think you're on the hook for implementing these diagnostics at all at the moment.  They should come from the underlying builtin.  The current PR has this just done in DXIL validation, but we also have this issue: https://github.com/microsoft/DirectXShaderCompiler/issues/7336

Reply:
I see. Well, I've written the extra tests, so I can merge them in and leave them as XFAIL for simon to modify as he sees fit.
Yes, I thought I was on the hook for 7336. I'll set the new tests with placeholdeer diagnostics to XFAIL.